File: app/src/main/java/com/bookshelfhub/bookshelfhub/adapters/slider/SliderAdapter.java
package com.bookshelfhub.bookshelfhub.adapters.slider;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.bookshelfhub.bookshelfhub.R;
import com.bookshelfhub.bookshelfhub.wrapper.imageloader.ImageLoader;
import com.bumptech.glide.Glide;
import com.smarteist.autoimageslider.SliderViewAdapter;

import java.util.ArrayList;
import java.util.List;

public class SliderAdapter extends
        SliderViewAdapter<SliderAdapterVH>  {

    private List<SliderItem> mSliderItems = new ArrayList<>();
    private final ImageLoader imageLoader;

    public SliderAdapter(ImageLoader imageLoader) {
        this.imageLoader = imageLoader;
    }

    public void renewItems(List<SliderItem> sliderItems) {
        this.mSliderItems = sliderItems;
        notifyDataSetChanged();
    }

    public void deleteItem(int position) {
        this.mSliderItems.remove(position);
        notifyDataSetChanged();
    }

    public void addItem(SliderItem sliderItem) {
        this.mSliderItems.add(sliderItem);
        notifyDataSetChanged();
    }


    @Override
    public SliderAdapterVH onCreateViewHolder(ViewGroup parent) {
        View inflate = LayoutInflater.from(parent.getContext()).inflate(R.layout.slider_item, null);
        return new SliderAdapterVH(inflate);
    }

    @Override
    public void onBindViewHolder(SliderAdapterVH viewHolder, int position) {
        SliderItem sliderItem = mSliderItems.get(position);

        viewHolder.firstTitle.setText(sliderItem.getFirstTitle());
        viewHolder.secondTitle.setText(sliderItem.getSecondTitle());
        viewHolder.textViewDescription.setText(sliderItem.getDescription());
        imageLoader.loadImageIntoView(sliderItem.getImageResourceId(), viewHolder.imageView);

    }

    @Override
    public int getCount() {
        return mSliderItems.size();
    }

}

File: app/src/main/java/com/bookshelfhub/bookshelfhub/adapters/viewpager/onboarding/SliderAdapter.kt
package com.bookshelfhub.bookshelfhub.adapters.viewpager.onboarding

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import com.bookshelfhub.bookshelfhub.R
import com.bookshelfhub.bookshelfhub.extensions.loadUnCompressed
import com.smarteist.autoimageslider.SliderViewAdapter
import java.util.*


class SliderAdapter: SliderViewAdapter<SliderAdapterVH>(){

    private val mSliderItems: MutableList<SliderItem> = ArrayList()


    fun addItem(sliderItem: SliderItem) {
        mSliderItems.add(sliderItem)
        notifyDataSetChanged()
    }

    @SuppressLint("InflateParams")
    override fun onCreateViewHolder(parent: ViewGroup): SliderAdapterVH {
        val inflate = LayoutInflater.from(parent.context).inflate(R.layout.slider_item, null)
        return SliderAdapterVH(inflate)
    }

    override fun onBindViewHolder(viewHolder: SliderAdapterVH?, position: Int) {
        if (viewHolder != null) {
            val sliderItem = mSliderItems[position]
            viewHolder.firstTitle.text = sliderItem.firstTitle
            viewHolder.secondTitle.text = sliderItem.secondTitle
            viewHolder.textViewDescription.text = sliderItem.description
            viewHolder.imageView.loadUnCompressed(sliderItem.resource)
        }
    }

    override fun getCount(): Int {
        return mSliderItems.size
    }

}